snippets/access-dom-element.md

```diff
@@ -0,0 +1,53 @@
+---
+title: Access DOM Element
+author: Alex Kulagin
+level: intermediate
+tags:
+- components
+- dom
+links:
+- https://angular.io/api/core/ElementRef
+- https://angular.io/guide/security
+---
+
+# Content
+When your 3rd party library needs access to dom element you can provide it with ElementRef. 
+```typescript
+@Directive({
+  selector: '[tooltip]'
+})
+export class TooltipDirective implements OnInit, OnDestroy {
+  private tooltip: Tooltip;
+  constructor(
+    private elementRef: ElementRef<HTMLElement>,
+  ) { }
+}
+```
+Be aware! Direct dom manipulations are bad practice
+
+# ComponentCode
+```typescript
+import { Directive, ElementRef, OnInit, OnDestroy } from '@angular/core';
+import Tooltip from 'tooltip.js';
```

Comment:
It will works on demo?

Reply:
Here is demo
https://stackblitz.com/edit/angular-tooltipjs-directive?file=src%2Fapp%2Ftooltip.directive.ts